tests/integration/17.SecurityTag.Tests.ps1

```diff
@@ -28,20 +28,38 @@ Describe -Name "SecurityTagAssignment" -Tag "Get" -Fixture {
         #We load the mod and establish connection to NSX Manager here.
 
         #Put any setup tasks in here that are required to perform your tests.  Typical defaults:
+        Write-Verbose -Verbose "Performing setup tasks for SecurityTag tests"
         Import-Module $pnsxmodule
         $script:DefaultNsxConnection = Connect-NsxServer -vCenterServer $PNSXTestVC -NsxServerHint $PNSXTestNSX -Credential $PNSXTestDefViCred -ViWarningAction "Ignore"
 
+        ## GUID to use as suffix for some temporary vSphere/NSX objects, so as to prevent any naming conflict with existing objects
+        $strSuffixGuid = [System.Guid]::NewGuid().Guid.Replace("-","")
+        Write-Verbose -Verbose "Using following GUID as suffix on temporary objects' names for ensuring unique object naming: '$strSuffixGuid'"
+
+        ## get some spot in which to create a couple of empty VMs, for use in tests (checking for SecurityTag assignments, for example)
+        $script:oVMHostToUse = Get-VMHost -State Connected | Get-Random
+        Write-Verbose -Verbose "Using VMHost '$oVMHostToUse' for temporary VM creation"
+        ## get the datastore mounted on this VMHost as readWrite (where .ExtensionData.Host.Key is this VMHost's ID, and where .ExtensionData.Host.MountInfo.AccessMode is "readWrite") and with the most FreespaceGB, on which to create some test VMs
+        $script:oDStoreToUse = $oVMHostToUse | Get-Datastore | Where-Object {$_.ExtensionData.Host | Where-Object {$_.Key -eq $oVMHostToUse.Id -and ($_.MountInfo.AccessMode -eq "readWrite")}} | Sort-Object -Property FreespaceGB -Descending | Select-Object -First 1
```

Comment:
use code from other test about datastore ?

Reply:
Howdy-  we could.  However, I wrote this "get a suitable datastore" code to help ensure success in a greater range of scenarios.  For example, the `$cl | get-datastore | select -first 1` code from another test suite failed in my testing environment as the first datastore happened to be an NFS datastore that is mounted `readOnly` (and, so, VM deployments to it failed, of course).

There is also probably more room for improvement here (this code just gets the readWrite datastore with the most free space on the given VMHost, but does not check for, "is the amount of free space sufficient?").  But, maybe that's for another day

Reply:
Yes, need some improvement.. 
like a option to choice the datastore for the first run (and store) like also for cluster...

Reply:
`$true` -- yes, or even parameters to the given test so that one can specify at invocation time the datastore/cluster/VMHost/etc. object(s) to use for the given tests if they want to use some specific resources..